apollo-ios-pagination/Sources/ApolloPagination/AnyGraphQLPager.swift

```diff
@@ -54,85 +56,84 @@ public class AnyGraphQLQueryPager<Model> {
-  @discardableResult public func subscribe(completion: @escaping (Output) -> Void) -> AnyCancellable {
-    guard let _subject else { return AnyCancellable({ }) }
-    let cancellable = _subject.compactMap({ $0 }).sink { result in
-      completion(result)
-    }
-    cancellable.store(in: &cancellables)
-    return cancellable
+  deinit {
+    pager.cancel()
+  }
+
+  /// Subscribe to the results of the pager, with the management of the subscriber being stored internally to the `AnyGraphQLQueryPager`.
+  /// - Parameter completion: The closure to trigger when new values come in. Guaranteed to run on the main thread.
+  public func subscribe(completion: @escaping @MainActor (Output) -> Void) {
+    publisher.sink { result in
+      Task { await completion(result) }
+    }.store(in: &cancellables)
   }
 
-  public func loadMore(
+  /// Load the next page, if available.
+  /// - Parameters:
+  ///   - cachePolicy: The Apollo `CachePolicy` to use. Defaults to `returnCacheDataAndFetch`.
+  ///   - completion: A completion block that will always trigger after the execution of this operation. Passes an optional error, of type `PaginationError`, if there was an internal error related to pagination. Does not surface network errors. Defaults to `nil`.
+  public func loadNext(
     cachePolicy: CachePolicy = .returnCacheDataAndFetch,
-    completion: (@MainActor () -> Void)? = nil
-  ) throws {
-    try pager.loadMore(cachePolicy: cachePolicy, completion: completion)
+    completion: ((Error?) -> Void)? = nil
+  ) {
+    pager.loadNext(cachePolicy: cachePolicy, completion: completion)
   }
 
+  /// Load the previous page, if available.
+  /// - Parameters:
+  ///   - cachePolicy: The Apollo `CachePolicy` to use. Defaults to `returnCacheDataAndFetch`.
+  ///   - completion: A completion block that will always trigger after the execution of this operation. Passes an optional error, of type `PaginationError`, if there was an internal error related to pagination. Does not surface network errors. Defaults to `nil`.
+  public func loadPrevious(
+    cachePolicy: CachePolicy = .returnCacheDataAndFetch,
+    completion: ((Error?) -> Void)? = nil
+  ) {
+    pager.loadPrevious(cachePolicy: cachePolicy, completion: completion)
+  }
+
+  /// Loads all pages.
+  /// - Parameters:
+  ///   - fetchFromInitialPage: Pass true to begin loading from the initial page; otherwise pass false.  Defaults to `true`.  **NOTE**: Loading all pages with this value set to `false` requires that the initial page has already been loaded previously.
+  ///   - completion: A completion block that will always trigger after the execution of this operation. Passes an optional error, of type `PaginationError`, if there was an internal error related to pagination. Does not surface network errors. Defaults to `nil`.
+  public func loadAll(
+    fetchFromInitialPage: Bool = true,
+    completion: ((Error?) -> Void)? = nil
```

Comment:
Weren't we going to change the completion blocks to specifically return a `PaginationError`?

Reply:
I'm surprised this didn't change, as I have it set that way in the internal GitHub version of this. The difficulties of maintaining this code in two places, I suppose 😅

Reply:
> Weren't we going to change the completion blocks to specifically return a `PaginationError`?